code/modules/client/preference_setup/loadout/gear_tweaks.dm

```diff
@@ -347,3 +347,39 @@
 	if(ValidTeslaLinks[metadata[7]])
 		var/t = ValidTeslaLinks[metadata[7]]
 		assembly.add_replace_component(null, PART_TESLA, new t(I))
+
+/datum/gear_tweak/charge_stick/get_contents(var/metadata)
+	return "Charge Stick: [metadata]"
+
+/datum/gear_tweak/charge_stick/get_default()
+	return "Partial"
+
+/datum/gear_tweak/charge_stick/get_metadata(var/user, var/list/metadata, title)
+	. = input(user, "Choose an entry.", CHARACTER_PREFERENCE_INPUT_TITLE, metadata) as null|anything in (list("None", "Partial", "Full"))
+	if(!.)
+		return metadata
+
+/datum/gear_tweak/charge_stick/tweak_item(var/obj/item/I, var/list/metadata)
```

Comment:
signature changed 
`/datum/gear_tweak/proc/tweak_item(var/user, var/obj/item/I, var/metadata)`
don't need to get user from loc anymore it's passed as arg